stl/inc/xlocnum

```diff
@@ -1292,18 +1292,23 @@ protected:
         _OutIt _Dest, ios_base& _Iosbase, _Elem _Fill, double _Val) const { // put formatted double to _Dest
         string _Buf;
         char _Fmt[8];
-        bool _Isfixed         = (_Iosbase.flags() & ios_base::floatfield) == ios_base::fixed;
-        streamsize _Precision = _Iosbase.precision() <= 0 && !_Isfixed ? 6 : _Iosbase.precision(); // desired precision
-        size_t _Bufsize       = static_cast<size_t>(_Precision);
+        const bool _Isfixed = (_Iosbase.flags() & ios_base::floatfield) == ios_base::fixed;
+        const bool _Ishex   = (_Iosbase.flags() & ios_base::floatfield) == (ios_base::fixed | ios_base::scientific);
+        const streamsize _Precision =
+            _Ishex ? (DBL_MANT_DIG + 2) / 4
+                   : _Iosbase.precision() <= 0 && !_Isfixed ? 6 : _Iosbase.precision(); // desired precision
+        size_t _Bufsize = static_cast<size_t>(_Precision);

```

Comment:
Stupid question: Is there a constant value we could use here for a reasonable maximum and replace `_Buf` with a stack array?

Reply:
According to my understanding, not without a significant amount of restructuring. The user could request `10000` precision, which would involve `sprintf_s` writing a bunch of zeroes after the decimal point.

(This *was* possible for `<charconv>` `general` precision - because that trims trailing zeroes, there is a maximum precision that can be calculated, and the result is reasonable to stack-allocate for 64-bit `double`.)